packages/salesforce-adapter/src/filters/extend_triggers_metadata.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2024 Salto Labs Ltd.
+ * Licensed under the Salto Terms of Use (the "License");
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+import {
+  BuiltinTypes,
+  Field,
+  InstanceElement,
+  isObjectType,
+  ObjectType,
+  CORE_ANNOTATIONS,
+  ReferenceExpression,
+  isAdditionOrModificationChange,
+  getChangeData,
+  Change,
+  ListType,
+} from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import { logger } from '@salto-io/logging'
+import { applyFunctionToChangeData, inspectValue } from '@salto-io/adapter-utils'
+import { RemoteFilterCreator } from '../filter'
+import {
+  apiNameSync,
+  buildElementsSourceForFetch,
+  ensureSafeFilterFetch,
+  isCustomObjectSync,
+  isInstanceOfTypeChangeSync,
+  isInstanceOfTypeSync,
+} from './utils'
+import { APEX_TRIGGER_METADATA_TYPE, INTERNAL_ID_FIELD } from '../constants'
+import SalesforceClient from '../client/client'
+import { SalesforceRecord } from '../client/types'
+
+const { toArrayAsync, awu } = collections.asynciterable
+const log = logger(module)
+
+const ON_DELETE_TRIGGER_TYPES = ['UsageBeforeDelete', 'UsageAfterDelete'] as const
+
+const ON_INSERT_TRIGGER_TYPES = ['UsageBeforeInsert', 'UsageAfterInsert'] as const
+
+const ON_UPDATE_TRIGGER_TYPES = ['UsageBeforeUpdate', 'UsageAfterUpdate'] as const
+
+export const TRIGGER_TYPE_FIELDS = [
+  ...ON_DELETE_TRIGGER_TYPES,
+  ...ON_INSERT_TRIGGER_TYPES,
+  ...ON_UPDATE_TRIGGER_TYPES,
+  'UsageAfterUndelete',
+] as const
+
```

Comment:
I think this is slightly more readable
```suggestion
enum TriggerType {
  UsageBeforeDelete = 'UsageBeforeDelete',
  UsageAfterDelete = 'UsageAfterDelete',
  UsageBeforeInsert = 'UsageBeforeInsert',
  UsageAfterInsert = 'UsageAfterInsert',
  UsageBeforeUpdate = 'UsageBeforeUpdate',
  UsageAfterUpdate = 'UsageAfterUpdate',
  UsageAfterUndelete = 'UsageAfterUndelete',
}

const TRIGGER_TYPE_FIELDS = Object.values(TriggerType);
```